### PR TITLE
Add height parameter to the Robotino3Platform

### PR DESCRIPTION
**Pull request by mael25**
This allows to change the height of the Robotino3Platform. One use case is when robots with different heights use this platform to put on top a ConveyorPlatform. Then the height of the Robotino3Platform can be set so that all ConveyorPlatform are at the same level.

robotino3platform_pr

### CODE REVIEW

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 1]
```diff
@@ -11,16 +11,21 @@ PROTO Robotino3Platform [
   field  SFString    name         "Platform for Robotino 3"      # Is `Solid.name`.
   field  SFString    model        "Festo - Robotino's platform"  # Is `Solid.model`.
   field  MFNode      extension    []                             # Extends the platform.
+  field  SFFloat     height       0.530447                       # Height of the platform.
 ]
 {
+  %{
+    HEIGHT = fields.height.value
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:18]:
```suggestion
    local height = fields.height.value
```

Comment (ad-daniel) [reply projects/robots/festo/robotino3/protos/Robotino3Platform.proto:18]: Here doesn't need to be global, and typically variable names are lowercase according to codestyle

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 2]
```diff
@@ -11,16 +11,21 @@ PROTO Robotino3Platform [
...
+
   Solid {
     translation IS translation
     rotation IS rotation
     name IS name
     model IS model
     children [
       DEF extensionSlot Transform {
-        translation 0.05 0 0.594
+        translation 0.05 0 %{= (0.594 - 0.530447) + HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:28]:
```suggestion
        translation 0.05 0 %{= (0.594 - 0.530447) + height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 3]
```diff
@@ -36,22 +41,22 @@ PROTO Robotino3Platform [
             point [
               0.000485 0.040875 -6e-06
               0.035661 -0.020055 -6e-06
-              0.035661 -0.020055 0.530447
-              0.000485 0.040875 0.530447
+              0.035661 -0.020055 %{= HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:44]:
```suggestion
              0.035661 -0.020055 %{= height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 4]
```diff
@@ -36,22 +41,22 @@ PROTO Robotino3Platform [
...
+              0.000485 0.040875 %{= HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:45]:
```suggestion
              0.000485 0.040875 %{= height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 5]
```diff
@@ -36,22 +41,22 @@ PROTO Robotino3Platform [
...
               -0.000512 0.040875 -6e-06
-              -0.000512 0.040875 0.530447
+              -0.000512 0.040875 %{= HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:47]:
```suggestion
              -0.000512 0.040875 %{= height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 6]
```diff
@@ -36,22 +41,22 @@ PROTO Robotino3Platform [
...
               -0.035657 -0.020037 -6e-06
-              -0.035657 -0.020037 0.530447
+              -0.035657 -0.020037 %{= HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:49]:
```suggestion
              -0.035657 -0.020037 %{= height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 7]
```diff
@@ -36,22 +41,22 @@ PROTO Robotino3Platform [
...
               -0.035178 -0.020867 -6e-06
-              -0.035178 -0.020867 0.530447
+              -0.035178 -0.020867 %{= HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:51]:
```suggestion
              -0.035178 -0.020867 %{= height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 8]
```diff
@@ -36,22 +41,22 @@ PROTO Robotino3Platform [
...
               0.035212 -0.020834 -6e-06
-              0.035212 -0.020834 0.530447
-              0.041455 -0.020055 0.530447
-              0.003382 0.045893 0.530447
-              -0.003409 0.045893 0.530447
-              -0.04145 -0.020037 0.530447
-              -0.038075 -0.025884 0.530447
-              0.038109 -0.025851 0.530447
+              0.035212 -0.020834 %{= HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:53]:
```suggestion
              0.035212 -0.020834 %{= height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 9]
```diff
@@ -36,22 +41,22 @@ PROTO Robotino3Platform [
...
+              0.041455 -0.020055 %{= HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:54]:
```suggestion
              0.041455 -0.020055 %{= height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 10]
```diff
@@ -36,22 +41,22 @@ PROTO Robotino3Platform [
...
+              0.003382 0.045893 %{= HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:55]:
```suggestion
              0.003382 0.045893 %{= height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 11]
```diff
@@ -36,22 +41,22 @@ PROTO Robotino3Platform [
...
+              -0.003409 0.045893 %{= HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:56]:
```suggestion
              -0.003409 0.045893 %{= height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 12]
```diff
@@ -36,22 +41,22 @@ PROTO Robotino3Platform [
...
+              -0.04145 -0.020037 %{= HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:57]:
```suggestion
              -0.04145 -0.020037 %{= height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 13]
```diff
@@ -36,22 +41,22 @@ PROTO Robotino3Platform [
...
+              -0.038075 -0.025884 %{= HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:58]:
```suggestion
              -0.038075 -0.025884 %{= height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 14]
```diff
@@ -36,22 +41,22 @@ PROTO Robotino3Platform [
...
+              0.038109 -0.025851 %{= HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:59]:
```suggestion
              0.038109 -0.025851 %{= height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto [hunk 15]
```diff
@@ -75,7 +80,7 @@ PROTO Robotino3Platform [
         }
       }
       DEF TABLE Transform {
-        translation 0 0 0.537
+        translation 0 0 %{= (0.537 - 0.530447) + HEIGHT }%
```
Comment (ad-daniel) [inline projects/robots/festo/robotino3/protos/Robotino3Platform.proto:83]:
```suggestion
        translation 0 0 %{= (0.537 - 0.530447) + height }%
```

File: projects/robots/festo/robotino3/protos/Robotino3Platform.proto (hunk 1 above)
Comment (mael25) [reply projects/robots/festo/robotino3/protos/Robotino3Platform.proto]: Thanks for the review!

**Comment by ad-daniel** [comment]
Looks good to me. It fails the tests due to unrelated issues, I think you can merge if you can't wait for the fix to cppcheck. Thank you.

**Review by ad-daniel — APPROVED** [review]

**Pull request merged**